projects/api/src/suppliers/suppliers.service.ts

```diff
@@ -14,12 +14,21 @@ class SuppliersService {
     try {
       const supplier: Supplier = await this.prisma.supplier.create({
         data: createSupplierDto,
+        include: {
+          country: true,
+        },
       });
       return supplier;
     } catch (error) {
-      if (error?.meta?.target?.includes('email')) {
+      if (!error.code) {
```

Comment:
hmmm maybe you can simply this batch of common error like:

`if (error?.meta?.target?.includes('email')) {
    if (!error.code) {
      throw new BadRequestException('Something went wrong');
    }
    if (error.code === 'P2002') {
      throw new BadRequestException('Supplier email already exists');
    }
  } else if (error.code === 'P2023') {
    throw new BadRequestException('Country ID not found');
  }

  throw new BadRequestException('Something went wrong');`

Reply:
hmmm I think this is not resolved... but it's not a big deal, I can leave for it...